app/models/json_builders/identity_json_builder.rb

```diff
@@ -0,0 +1,104 @@
+    @parsed_publisher_name = URL_REGULAR_EXPRESSION.match(@publisher_name)
+    find_channel_detail
+  end
+
+  def find_channel_detail
+    @channel_detail = if @parsed_publisher_name.present? && @parsed_publisher_name[1] == Channel::YOUTUBE
+      YoutubeChannelDetails.find_by(youtube_channel_id: @parsed_publisher_name[3])
+    elsif @parsed_publisher_name.present? && @parsed_publisher_name[1] == Channel::TWITCH
+      TwitchChannelDetails.find_by(twitch_channel_id: @parsed_publisher_name[3])
+    else
+      SiteChannelDetails.find_by(brave_publisher_id: @publisher_name)
+    end
+  end
+
+  def build
+    @json = if @channel_detail.nil?
+      # @errors << "Channel not found"
+      build_site_identity_json
+    elsif @channel_detail.is_a?(YoutubeChannelDetails)
+      build_youtube_identity_json
+    elsif @channel_detail.is_a?(TwitchChannelDetails)
+      build_twitch_identity_json
+    elsif @channel_detail.is_a?(SiteChannelDetails)
+      build_site_identity_json
+    end
+    self
+  end
+
+  def result
+    @json
+  end
+
+  def build_youtube_identity_json
+    Jbuilder.encode do |json|
+      json.publisher      @publisher_name
+      json.publisherType  'provider'
+      json.providerName   @parsed_publisher_name[1]
+      json.providerSuffix @parsed_publisher_name[2]
+      json.providerValue  @parsed_publisher_name[3]
+      json.URL            "https://youtube.com/#{@parsed_publisher_name[2]}/#{@parsed_publisher_name[3]}"
+      json.TLD            @publisher_name.split(':')[0]
+      json.SLD            @publisher_name
+      json.RLD            @parsed_publisher_name[3]
+      json.QLD            ''
+      json.properties     do
+        build_properties(json)
+      end
+    end
+  end
+
+  def build_site_identity_json
+    Jbuilder.encode do |json|
+      json.publisher      @publisher_name
+      json.SLD            @publisher_name
+      json.RLD            ''
+      json.QLD            ''
+      json.URL            @publisher_name
+      json.properties     do
+        build_properties(json)
+      end
+    end
+  end
+
+  def build_twitch_identity_json
+    Jbuilder.encode do |json|
+      json.publisher      @publisher_name
+      json.publisherType  'provider'
+      json.providerName   @parsed_publisher_name[1]
+      json.providerSuffix @parsed_publisher_name[2]
+      json.providerValue  @parsed_publisher_name[3]
+      json.TLD            @publisher_name.split(':')[0]
+      json.SLD            @publisher_name
+      json.RLD            @parsed_publisher_name[3]
+      json.QLD            ''
+      json.properties     do
+        build_properties(json)
+      end
+    end
+  end
+
+  def build_properties(json)
+    require 'publishers/excluded_channels'
+    if @channel_detail.present? && @channel_detail.channel.verified?
+      json.verified true
+      # (Albert Wang): To satisfy backwards compatibility in Ledger's v3.identity
+      json.timestamp (@channel_detail.channel.updated_at.to_i << 32)
```

Comment:
What is the purpose of the timestamp? A note on the logic would be good here as well. That would help me understand why it's based on `updated_at` vs `verified_at`.

The timestamp logic doesn't seem to be tested. The reason for the bit shift isn't apparent so a note would be helpful as well as a test.